RadarSDK/RadarNotificationHelper.m

```diff
@@ -68,4 +74,149 @@ + (void)showNotificationsForEvents:(NSArray<RadarEvent *> *)events {
     }
 }
 
++ (void)swizzleNotificationCenterDelegate {
+    id<UNUserNotificationCenterDelegate> delegate = UNUserNotificationCenter.currentNotificationCenter.delegate;
+    if (!delegate) {
+        NSLog(@"Error: UNUserNotificationCenter delegate is nil.");
+        return;
+    }
+    Class class = [UNUserNotificationCenter.currentNotificationCenter.delegate class];
+    SEL originalSelector = @selector(userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:);
+    SEL swizzledSelector = @selector(swizzled_userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:);
+
+    Method originalMethod = class_getInstanceMethod(class, originalSelector);
+    Method swizzledMethod = class_getInstanceMethod([self class], swizzledSelector);
+
+    if (originalMethod && swizzledMethod) {
+        BOOL didAddMethod = class_addMethod(class,
+                                            swizzledSelector,
+                                            method_getImplementation(swizzledMethod),
+                                            method_getTypeEncoding(swizzledMethod));
+        if (didAddMethod) {
+            Method newSwizzledMethod = class_getInstanceMethod(class, swizzledSelector);
+            method_exchangeImplementations(originalMethod, newSwizzledMethod);
+        } else {
+            method_exchangeImplementations(originalMethod, swizzledMethod);
+        }
+    } else {
+        NSLog(@"Error: Methods not found for swizzling.");
+    }
+}
+
+- (void)swizzled_userNotificationCenter:(UNUserNotificationCenter *)center
+           didReceiveNotificationResponse:(UNNotificationResponse *)response
+                    withCompletionHandler:(void (^)(void))completionHandler {
+
+    if ([response.notification.request.identifier hasPrefix:@"radar_"]) {
+        [[RadarLogger sharedInstance]
+                        logWithLevel:RadarLogLevelDebug
+                            message:[NSString stringWithFormat:@"Getting conversion from notification tap"]];
+        [Radar logConversionWithNotification:response.notification.request eventName:@"opened_app" conversionSource:@"radar_notification" deliveredAfter:nil];
+    } else {
+        [Radar logConversionWithNotification:response.notification.request eventName:@"opened_app" conversionSource:@"notification" deliveredAfter:nil];
+    }
+    // do we still want to log the normal app open event?
+    [RadarSettings updateLastAppOpenTime];
```

Comment:
Note to self to check this against the `opened_app` logic.

Reply:
yeah this is right, since we are now having the open notification be a form of the `opened_app` event, we do not want to call the code inside `logOpenedAppConversion`